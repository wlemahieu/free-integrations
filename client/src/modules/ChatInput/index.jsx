import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { Col, Input, Row } from 'antd';
const { Search } = Input;

class ChatInput extends PureComponent {
	render() {
		const { disabledSubmit, input } = this.props;
		return (
			<Row className="centered" style={{ paddingTop: '40px' }}>
				<Col span={12} offset={6}>
				<Search
					placeholder="What's on your mind?"
					enterButton
					size="large"
					onChange={e => this.props.updateInput(e.target.value)}
					onSearch={e => this.props.onSearch(input)}
					disabled={disabledSubmit}
					value={input}
				/>
				</Col>
			</Row>
		);
	}
}

ChatInput.propTypes = {
	disabledSubmit: PropTypes.bool,
	input: PropTypes.string,
	onSearch: PropTypes.func
};

ChatInput.defaultProps = {
	disabledSubmit: false,
	input: '',
	onSearch: () => {}
};

export default ChatInput;
