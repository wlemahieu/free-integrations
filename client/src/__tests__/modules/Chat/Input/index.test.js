import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ChatInput from 'modules/Chat/Input';

const props = {
  disabledSubmit: false,
  input: ''
};

let wrapper;
beforeEach(() => {
  wrapper = mount(
    <ChatInput
      disabledSubmit={props.disabledSubmit}
      input={props.input}
    />
  );
});

describe('<ChatInput />', () => {
  it('matches snapshot', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it('renders single row className="centered" gutter=0', () => {
    const Rows = wrapper.find('Row');
    const { className, gutter } = Rows.props();
    expect(Rows).toHaveLength(1);
    expect(className).toEqual('centered');
    expect(gutter).toEqual(0);
  });
  it('renders single column span=12 offset=6', () => {
    const Cols = wrapper.find('Col');
    const { span, offset } = Cols.props();
    expect(Cols).toHaveLength(1);
    expect(span).toEqual(12);
    expect(offset).toEqual(6);
  });
  it('renders single large search input with specific placeholder verbiage', () => {
    const Search = wrapper.find('Search');
    const { placeholder, size } = Search.props();
    expect(Search).toHaveLength(1);
    expect(placeholder).toBe('Say something to Rose Watson');
    expect(size).toBe('large');
  });
});
