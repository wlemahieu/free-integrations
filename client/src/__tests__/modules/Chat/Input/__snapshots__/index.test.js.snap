// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChatInput /> matches snapshot 1`] = `
<ChatInput
  disabledSubmit={false}
  input=""
  onSearch={[Function]}
  updateInput={[Function]}
>
  <Row
    className="centered"
    gutter={0}
    style={
      Object {
        "paddingTop": "40px",
      }
    }
  >
    <div
      className="ant-row centered"
      style={
        Object {
          "paddingTop": "40px",
        }
      }
    >
      <Col
        offset={6}
        span={12}
      >
        <div
          className="ant-col ant-col-12 ant-col-offset-6"
          style={Object {}}
        >
          <Search
            autoFocus={true}
            disabled={false}
            enterButton={true}
            onChange={[Function]}
            onSearch={[Function]}
            placeholder="Say something to Rose Watson"
            size="large"
            value=""
          >
            <Input
              addonAfter={
                <Button
                  block={false}
                  className="ant-input-search-button"
                  disabled={false}
                  ghost={false}
                  htmlType="button"
                  loading={false}
                  onClick={[Function]}
                  size="large"
                  type="primary"
                >
                  <Icon
                    type="search"
                  />
                </Button>
              }
              autoFocus={true}
              className="ant-input-search ant-input-search-enter-button ant-input-search-large"
              disabled={false}
              onChange={[Function]}
              onPressEnter={[Function]}
              placeholder="Say something to Rose Watson"
              prefixCls="ant-input"
              size="large"
              type="text"
              value=""
            >
              <ClearableLabeledInput
                addonAfter={
                  <Button
                    block={false}
                    className="ant-input-search-button"
                    disabled={false}
                    ghost={false}
                    htmlType="button"
                    loading={false}
                    onClick={[Function]}
                    size="large"
                    type="primary"
                  >
                    <Icon
                      type="search"
                    />
                  </Button>
                }
                autoFocus={true}
                className="ant-input-search ant-input-search-enter-button ant-input-search-large"
                disabled={false}
                element={
                  <input
                    autoFocus={true}
                    className="ant-input ant-input-lg"
                    disabled={false}
                    onChange={[Function]}
                    onKeyDown={[Function]}
                    placeholder="Say something to Rose Watson"
                    type="text"
                    value=""
                  />
                }
                handleReset={[Function]}
                inputType="input"
                onChange={[Function]}
                onPressEnter={[Function]}
                placeholder="Say something to Rose Watson"
                prefixCls="ant-input"
                size="large"
                type="text"
                value=""
              >
                <span
                  className="ant-input-search ant-input-search-enter-button ant-input-search-large ant-input-group-wrapper ant-input-group-wrapper-lg"
                >
                  <span
                    className="ant-input-wrapper ant-input-group"
                  >
                    <input
                      autoFocus={true}
                      className="ant-input ant-input-lg"
                      disabled={false}
                      onChange={[Function]}
                      onKeyDown={[Function]}
                      placeholder="Say something to Rose Watson"
                      style={null}
                      type="text"
                      value=""
                    />
                    <span
                      className="ant-input-group-addon"
                    >
                      <Button
                        block={false}
                        className="ant-input-search-button"
                        disabled={false}
                        ghost={false}
                        htmlType="button"
                        key="enterButton"
                        loading={false}
                        onClick={[Function]}
                        size="large"
                        type="primary"
                      >
                        <Wave>
                          <button
                            className="ant-btn ant-input-search-button ant-btn-primary ant-btn-lg"
                            disabled={false}
                            onClick={[Function]}
                            type="button"
                          >
                            <Icon
                              key=".0"
                              type="search"
                            >
                              <LocaleReceiver
                                componentName="Icon"
                              >
                                <i
                                  aria-label="icon: search"
                                  className="anticon anticon-search"
                                >
                                  <IconReact
                                    className=""
                                    type="search-o"
                                  >
                                    <svg
                                      aria-hidden="true"
                                      className=""
                                      data-icon="search"
                                      fill="currentColor"
                                      focusable="false"
                                      height="1em"
                                      key="svg-search"
                                      viewBox="64 64 896 896"
                                      width="1em"
                                    >
                                      <path
                                        d="M909.6 854.5L649.9 594.8C690.2 542.7 712 479 712 412c0-80.2-31.3-155.4-87.9-212.1-56.6-56.7-132-87.9-212.1-87.9s-155.5 31.3-212.1 87.9C143.2 256.5 112 331.8 112 412c0 80.1 31.3 155.5 87.9 212.1C256.5 680.8 331.8 712 412 712c67 0 130.6-21.8 182.7-62l259.7 259.6a8.2 8.2 0 0 0 11.6 0l43.6-43.5a8.2 8.2 0 0 0 0-11.6zM570.4 570.4C528 612.7 471.8 636 412 636s-116-23.3-158.4-65.6C211.3 528 188 471.8 188 412s23.3-116.1 65.6-158.4C296 211.3 352.2 188 412 188s116.1 23.2 158.4 65.6S636 352.2 636 412s-23.3 116.1-65.6 158.4z"
                                        key="svg-search-svg-0"
                                      />
                                    </svg>
                                  </IconReact>
                                </i>
                              </LocaleReceiver>
                            </Icon>
                          </button>
                        </Wave>
                      </Button>
                    </span>
                  </span>
                </span>
              </ClearableLabeledInput>
            </Input>
          </Search>
        </div>
      </Col>
    </div>
  </Row>
</ChatInput>
`;
